learn/os_setup/install_gfortran.md

```diff
@@ -0,0 +1,109 @@
+Binaries compiled in Cygwin are 'Cygwin' binaries linked against `cygwin.dll` and will hence only execute within the Cygwin environment. Binaries compiled in WSL are Linux binaries and therefore require the WSL environment to run on Windows.
+
+## Linux
+
+
+### Debian-based (Debian, Ubuntu, Mint, etc...)
+Check whether you have gfortran already installed
+```bash
+which gfortran
+```
+If nothing is returned then gfortran is not installed.
+To install gfortran type:
+```bash
+sudo apt-get install gfortran
+```
+to check what version was installed type:
+```bash
+gfortran --version
+```
+You can install multiple versions up to version 9 just by typing immidiately after gfortran[-number] e.g.:
+```bash
+sudo apt-get install gfortran-7 
+```
+To install the latest version 10 you need first to add / update the following repository and then install: 
+```bash
+add-apt-repository ppa:ubuntu-toolchain-r/test
+
+apt update
+
+apt install gfortran-10
+```
+Finally, you can switch between different versions or set the default one with the **update-alternatives** ([see manpage](http://manpages.ubuntu.com/manpages/trusty/man8/update-alternatives.8.html#:~:text=update%2Dalternatives%20creates%2C%20removes%2C,system%20at%20the%20same%20time.)). There are plenty online tutorials on how to use this feature.
+
+### RPM-based (Red Hat Linux, CentOS, Fedora, openSuse, Mandrake Linux)
+```bash
+yum install gcc-gfortran
+```
+
+### Arch-based (Arch Linux, Antergos, Manjaro, etc...)
+```bash
+sudo pacman -S gcc-fortran
+```
+
+## MacOS
+### Xcode
+If you have Xcode installed, open a terminal window and type:
+```bash
+xcode-select --install
+```
+### Binaries
+Check this github repo: [fxcoudert/gfortran-for-macOS](https://github.com/fxcoudert/gfortran-for-macOS/releases) to directly install binaries.
+### Homebrew
+```bash
+brew install gcc
+```
+### Fink
+GNU-gcc Package [link](https://pdb.finkproject.org/pdb/browse.php?summary=GNU+Compiler+Collection+Version)
+### MacPorts
+Search for available gcc versions:
+```bash
+port search gcc
+```
+Install a gcc version:
+```bash
+sudo port install gcc10
+```
+---
+## OpenCoarrays
+
+The best way to describe [OpenCoarrays](http://www.opencoarrays.org/) is to directly quote from the official site: *OpenCoarrays is an open-source software project that produces an application binary interface (ABI) used by the GNU Compiler Collection (GCC) Fortran front-end to build executable programs that leverage the parallel programming features of Fortran 2018*. Therefore, since OpenCoarrays is not a separate compiler, we include it here, under gfortran.
+
+The difference with gfortran is that, while with gfortran one can write perfectly valid code involving coarrays features, the generated code will only run in a single thread. OpenCoarrays allows to run code in parallel in a similar way as with MPI:
+```bash
+$ cafrun -n [number_of_threads] [executable_name]
```

Comment:
Few edits here. We don't use `$` elsewhere so I remove it from here. Second, I don't think "threads" is quite accurate, I suggest "images". Third, square brackets often denote optional arguments, so I think angle brackets are more appropriate.

```suggestion
cafrun -n <number_of_images> <executable_name>
```